### ISSUE TITLE
💅noDelete suggest assigning undefined instead of delete for process.env vars

### ISSUE THREAD

**Issue by robrat** [biomejs/biome#4093]
### Environment information

```bash
Assigning undefined to process.env vars does not work since it will be converted to a string ('undefined'):

$ node
Welcome to Node.js v20.17.0.
Type ".help" for more information.
> process.env.FOO
undefined
> process.env.FOO = 'bar'
'bar'
> process.env.FOO = undefined
undefined
> typeof process.env.FOO
'string'
> delete process.env.FOO
true
> typeof process.env.FOO
'undefined'

See https://nodejs.org/docs/latest-v20.x/api/process.html#processenv for additional details. It also states "Use delete to delete a property from process.env." Please also note that an assignment of undefined is deprecated and future versions will throw an error.
```

### Rule name

lint/performance/noDelete

### Playground link

https://biomejs.dev/playground/?code=ZABlAGwAZQB0AGUAIABwAHIAbwBjAGUAcwBzAC4AZQBuAHYALgBGAE8ATwA7AA%3D%3D

### Expected result

The rule should not suggest assigning undefined to process.env vars, but instead an assignment of undefined should be reported as error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
> but instead an assignment of undefined should be reported as error.

Have some motivation about that?
Also note that this seems be out of scope for this rule.

**Comment by robrat** [comment]
Because the suggestion of using `process.env.FOO = undefined` will result in wrong code, therefore the rule should at least exclude `process.env`.

I don't know how the rules are working in detail, but an assignment of `undefined` to a `process.env` property behaves differently compared to other assignments and therefore it is most likely a bug.

**Comment by Conaclos** [comment]
Deleting a `process.env` looks also wrong to me. This should be treated as a readonly variable.

**Comment by ematipico** [comment]
> Deleting a `process.env` looks also wrong to me. This should be treated as a readonly variable.

The official docs endorse that, whether we agree or not.

**Comment by robrat** [comment]
The documentation clearly states

> Use delete to delete a property from process.env.
>
> ```js
> const { env } = require('node:process');
>
> env.TEST = 1;
> delete env.TEST;
> console.log(env.TEST);
> // => undefined
> ```

I need this functionality to reset variables in unit tests of AWS Lambda functions.

**Comment by galabra** [comment]
@ematipico So the expected behavior here is that `delete process.env.FOO` won't raise an error?

**Comment by ematipico** [comment]
Yes, and even the following snippets shouldn't raise an error

```js
const { env } = require('node:process');
delete env.TEST;
```


```js
const { env } = require('process');
delete env.TEST;
```

**Comment by Cajuteq** [comment]
I came here because this rules made me change the following line and broke a file upload functionality
```
<<< delete headers['Content-Type'];
>>> headers['Content-Type'] = undefined;
```
I'm not blaming anyone, I should have been more careful applying the suggested fix, but I feel like this suggestion is misleading

**Comment by ematipico** [comment]
I started working on enhancing this rule
